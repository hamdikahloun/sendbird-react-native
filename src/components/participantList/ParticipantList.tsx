import React from 'react';
import {FlatList, StyleSheet, View} from 'react-native';
import {User} from 'sendbird';
import {Participant} from '../participant';

type ParticipantListProps = {
  participants: User[];
  onParticipantPressed?: (participant: User) => void;
};

/**
 * ParticipantList - The message Participant component renders a list of Participants.
 *
 * See:
 *
 * @example
 *
 * <ParticipantList
 *  participants={participants}
 *  onParticipantPressed={onParticipantPressed}
 * />
 */
export const ParticipantList = ({
  participants,
  onParticipantPressed,
}: ParticipantListProps) => {
  const renderItem = ({item}: {item: User}) => (
    <Participant
      onParticipantPressed={onParticipantPressed}
      participant={item}
    />
  );
  return (
    <View style={styles.container}>
      <FlatList
        horizontal
        data={participants}
        renderItem={renderItem}
        keyExtractor={item => item.userId}
        showsHorizontalScrollIndicator={false}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
    paddingBottom: 8,
    backgroundColor: '#f3F3F3',
  },
});
