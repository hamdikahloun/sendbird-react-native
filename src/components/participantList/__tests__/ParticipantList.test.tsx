import React from 'react';
import {render} from '@/../testing';
import {ParticipantList} from '../ParticipantList';

describe('ParticipantList', () => {
  it('should render correctly', async () => {
    const onParticipantPressed = jest.fn();
    const screen = render(
      <ParticipantList
        onParticipantPressed={onParticipantPressed}
        participants={[]}
      />,
    ).toJSON();
    expect(screen).toMatchSnapshot();
  });
});
