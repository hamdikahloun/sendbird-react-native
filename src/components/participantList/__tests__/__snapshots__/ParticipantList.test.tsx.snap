// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantList should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f3F3F3",
      "paddingBottom": 8,
      "paddingHorizontal": 16,
    }
  }
>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
